enhancements/windows-containers/bring-your-own-windows-host.md

```diff
@@ -0,0 +1,196 @@
+_windows-instances_ ConfigMap and node objects. Any change will cause WMCO to
+compare the list of instances with the set of existing Windows nodes. The
+comparison will be based on the instance and node's IP addresses. WMCO will
+then react in the following manner:
+* If a new entry is found i.e. the list of nodes does not contain one that map
+  to the instances' IP address, the instance will be configured by WMCO. The
+  configuration process will be identical to configuring instances added using
+  a MachineSet in an IPI cluster except in addition, WMCO will also have to
+  handle CSR approvals.
+* If the list of nodes has IP addresses that do not map to any in the list of
+  instances, WMCO will perform an uninstall operation on the node. This will
+  involve:
+  * Drain and cordon the node
+  * Stop and remove:
+    * Monitoring and logging services
+    * kube-proxy service
+    * hybrid-overlay-node service
+    * kubelet service
+  * Delete the payload binaries
+* If the list of nodes has IP addresses that do not map to any in the list of
+  instances, and WMCO is unable to access the instance, it will assume that
+  the node in question is not managed by it.
+
+The status of these operations will be reflected by events in the
+_openshift-windows-machine-config-operator_ namespace and will be associated
+with the _windows-instances_ ConfigMap.
+
+The upgrade process for BYOH nodes will be identical to upgrading Windows nodes
+added using a MachineSet in an IPI cluster.
+
+### Test Plan
+
+The existing set of WMCO end to end tests will be run against already created
+Windows instances. This can be achieved by creating a Windows Machine that does
+not have the _machine.openshift.io/os-id: Windows_ label. Once this Machine has
+been created the Machine API and cluster provider controllers should be
+disabled to prevent them from reacting. Then the created Machine's IP can then
+be added to / removed from the _windows-instances_ ConfigMap which will cause
+WMCO to react.
+
+### Graduation Criteria
+
+This enhancement will start as GA.
+
+### Upgrade / Downgrade Strategy
+
+The upgrade process for BYOH nodes will be different to that of
+[upgrading Windows nodes](https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/windows-machine-config-operator-upgrades.md)
+added using a MachineSet in an IPI cluster. In the MachineSet case, Machines
+are deleted in lieu of upgrading where as here the Windows instances will have
+to be reconfigured. The reconfiguring process will include:
+* Drain and cordon the node
+* Stop, copy over new binaries and update:
+  * Monitoring and logging services
+  * kube-proxy service
+  * hybrid-overlay-node service
+  * kubelet service
+* Delete unused services and binaries if we move away from using any of them
+* Uncordon the node
+
+### Version Skew Strategy
+
+We plan to maintain kubelet major version parity with the Linux counterpart.
```

Comment:
Even if Linux in OpenShift moves to N-2? I think given you're managed via OLM rather than MachineConfig you could remain N-1 even if Linux nodes move to N-2 in the future. It seemed like @derekwaynecarr had specific concerns that Windows Kubelet still had a way to go to reach parity with Linux as far as version skew policy.

Maybe for now lets be explicit and say N-1 only?